nbgrader/tests/test_nbextension.py

```diff
@@ -0,0 +1,200 @@
+import subprocess as sp
+import tempfile
+import os
+import shutil
+
+from nose.tools import assert_equal
+
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support.ui import Select
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.common.keys import Keys
+
+class TestCreateAssignmentNbExtension(object):
+
+    @classmethod
+    def setup_class(cls):
+        cls.tempdir = tempfile.mkdtemp()
+        cls.ipythondir = tempfile.mkdtemp()
+        cls.origdir = os.getcwd()
+        os.chdir(cls.tempdir)
+
+        cls.nbserver = sp.Popen([
+            "ipython", "notebook",
+            "--ipython-dir", cls.ipythondir,
+            "--no-browser",
+            "--port", "9000"], stdout=sp.PIPE, stderr=sp.STDOUT)
+
+    @classmethod
+    def teardown_class(cls):
+        cls.nbserver.kill()
+
+        os.chdir(cls.origdir)
+        shutil.rmtree(cls.tempdir)
+        shutil.rmtree(cls.ipythondir)
+
+    def setup(self):
+        shutil.copy(os.path.join(os.path.dirname(__file__), "files", "blank.ipynb"), "blank.ipynb")
+        self.browser = webdriver.PhantomJS()
```

Comment:
I didn't try any of the other drivers as I didn't know whether they would work on travis, though @Carreau tells me that you can use saucelabs with travis to test the other drivers. I did initially try the firefox driver locally at the very beginning, and that seemed to work fine out of the box, so I would expect it to be pretty easy to switch drivers.